lib/charms/hydra/v0/oauth.py

```diff
@@ -733,12 +729,14 @@ def _get_client_config_from_relation_data(self, event: RelationChangedEvent) ->
     def _get_secret_label(self, relation: Relation) -> str:
         return f"client_secret_{relation.id}"
 
-    def _on_relation_broken(self, event: RelationBrokenEvent) -> None:
-        # Workaround for https://github.com/canonical/operator/issues/888
-        self._pop_relation_data(event.relation.id)
+    # There is no way to tell if this event was emittede because the relation was removed or if one of
+    # the applications was scaled down. Until this is fixed, we don't delete the client.
```

Comment:
Question: when scaled down, does only the departing unit receive this event? If that's the case maybe we could just check for leadership before deleting the client?

Reply:
The docs have changed and I am struggling to find it, but afaict only the departing unit gets the event. Still if someone deletes the leader unit, all clients would be deleted. Even if juju will not do this on it's own, someone could do remove that unit manually. Not sure if we are fine with this.